r/tests/testthat/test-dataset-dplyr.R

```diff
@@ -80,26 +80,29 @@ test_that("filter() on timestamp columns", {
     df1[5:10, c("ts")],
   )
 
-  # Now with Date
+  # Now with bare string date
+  skip("Implement more aggressive implicit casting for scalars (ARROW-11402)")
   expect_equal(
     ds %>%
-      filter(ts >= as.Date("2015-05-04")) %>%
+      filter(ts >= "2015-05-04") %>%
       filter(part == 1) %>%
       select(ts) %>%
       collect(),
     df1[5:10, c("ts")],
   )
 
-  # Now with bare string date
-  skip("Implement more aggressive implicit casting for scalars (ARROW-11402)")
+  # string processing requires RE2 library (not available on Windows with R 3.6)
+  skip_if_not_available("re2")
```

Comment:
I moved this test at the end of the test block since the `as.Date()` binding now depends on RE2 (not available in R 3.6 on Windows).